integration-tests/test/test_bonding_validators.py

```diff
@@ -39,61 +35,82 @@ def test_heterogenous_validators(command_line_options: CommandLineOptions, rando
     }
 
     bonded_validator_map = {
-        BOOTSTRAP_KEY: 200,
-        BONDED_VALIDATOR_KEY: 300
+        BOOTSTRAP_KEY: 10000000,
+        BONDED_VALIDATOR_KEY: 10000000
     }
 
+    epoch_length = 4
+    quarantine_length = 2
     with conftest.testing_context(command_line_options, random_generator, docker_client, validator_bonds_dict=bonded_validator_map, wallets_dict=genesis_vault) as context, \
-        ready_bootstrap_with_network(context=context) as bootstrap_node, \
-        bootstrap_connected_peer(context=context, bootstrap=bootstrap_node, name='bonded-validator', private_key=BONDED_VALIDATOR_KEY) as bonded_validator:
+        ready_bootstrap_with_network(context=context, synchrony_constraint_threshold=0, epoch_length=epoch_length, quarantine_length=quarantine_length) as bootstrap_node, \
+        bootstrap_connected_peer(context=context, bootstrap=bootstrap_node, name='bonded-validator', private_key=BONDED_VALIDATOR_KEY, epoch_length=epoch_length, quarantine_length=quarantine_length) as bonded_validator:
             wait_for_peers_count_at_least(context, bonded_validator, 1)
+            # genesis block number is 0
 
             contract_path = '/opt/docker/examples/hello_world_again.rho'
+
+            # block number 1
             bonded_validator.deploy(contract_path, BOOTSTRAP_KEY)
             latest_block_hash = bonded_validator.propose()
+
             # assure the joining validator is not bonded
             block_info = bonded_validator.show_block_parsed(latest_block_hash)
             assert block_info.bonds.get(JOINING_VALIDATOR_KEY.get_public_key().to_hex()) is None
 
             with bootstrap_connected_peer(context=context, bootstrap=bootstrap_node, name='joining-validator', private_key=JOINING_VALIDATOR_KEY) as joining_validator:
                 wait_for_peers_count_at_least(context, bonded_validator, 2)
+
                 # new joining validator can not propose before bonding
                 with pytest.raises(NonZeroExitCodeError):
                     joining_validator.deploy(contract_path, JOINING_VALIDATOR_KEY)
                     joining_validator.propose()
 
+                # block number 2
```

Comment:
What causes the block number to change here?

Reply:
https://github.com/rchain/rchain/pull/2754/files#diff-73712dc027d17012dbd7e2db68f27779L67-R78